src/Servers/Kestrel/samples/QuicSampleClient/Program.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Buffers;
+using System.Net;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Connections.Features;
+using Microsoft.AspNetCore.Server.Kestrel.Https;
+using Microsoft.AspNetCore.Server.Kestrel.Transport.MsQuic;
+
+namespace QuicSampleClient
+{
+    class Program
+    {
+        static async Task Main(string[] args)
+        {
+            var transportOptions = new MsQuicTransportOptions()
+            {
+                Alpn = "QuicTest",
+                RegistrationName = "Quic-AspNetCore-client",
+                Certificate = CertificateLoader.LoadFromStoreCert("localhost", StoreName.My.ToString(), StoreLocation.CurrentUser, true),
+                IdleTimeout = TimeSpan.FromHours(1),
+            };
+
+            var transportContext = new MsQuicTransportContext(null, null, transportOptions);
+            var factory = new MsQuicConnectionFactory(transportContext);
+            var connectionContext = await factory.ConnectAsync(new IPEndPoint(IPAddress.Loopback, 5555));
+            var streamFeature = connectionContext.Features.Get<IStreamListener>();
+            var streamContext = await streamFeature.StartBidirectionalStreamAsync();
+
+            Console.CancelKeyPress += new ConsoleCancelEventHandler((sender, args) =>
+            {
+                streamContext.Transport.Output.Complete();
```

Comment:
Complete after the loop and wait on the Console like this https://github.com/davidfowl/BedrockTransports/blob/4f2bc78b14e69fb7420919cdedab300a9ec6a3ea/ServerApplication/Program.cs#L73-L75

Reply:
Still has the same issue of calling Complete in the middle of the loop. How do you cause the loop to exit here normally? Use CancelPendingRead?

Reply:
To gracefully exit even when there's write backpressure, you probably want to look into calling CancelPendingWrite too. Also, does Console.ReadLine() unblock on Ctrl+C? You probably want to look into how to interrupt the ReadLine call and handle the null return value it might produce.

Or maybe, since this is just a sample, we just never complete the PipeWriter and just call Environment.Exit(0) in the CancelKeyPress event handler.

Reply:
I just pushed a change to call CancelPendingRead and CancelPendingWrite.